### PR TITLE
Separate some test cases that need to be tested on the cuda backend

### PR DESCRIPTION
**Pull request by wenhuiNi**
cuda_tests(total num:28) = [`'kernel-launch', 'backprop', 'device_info', 'macro', 'driverMem', 'driverCtx', 'driverTex',  'driverDevice', 'driverStreamAndEvent', 'deviceAPI', 'free-queries', 'deviceProp', 'gridThreads',  'test_shared_memory', 'activemask', 'result_type_overload', 'defaultStream', 'group-local-memory', 'constant_attr', 'pointer_attributes', 'perm_byte', 'feature_profiling', 'math-bf16-conv',  'math-bfloat16',  'math-drcp', 'math-exec', 'math-habs', 'volatile-vec'`] 

 cuda_help_function(total num:26) = [`'array_copy', 'assign_device_vector', 'move_device_vector', 'atomic_add_float', 'atomic_fetch_compare_inc', 'async_exception', 'dpct_device_info', 'max', 'memcpy3D', 'memcpy3D_2', 'global_memory', 'global_memory_usm', 'shared_memory', 'shared_memory_usm', 'test_default_queue_2', 'math_function', 'memory_async_dpct_free',  'dpct_dpct_align_and_dpct_inline', 'dpct_dpct_compatibility_temp', 'atomic_atomic_fetch_sub', 'atomics_usm_verification',  'device_util', 'device_dev_mgr', 'device_device_ext', 'device_device_info', 'atomic_compare_exchange_strong'`]

### CODE REVIEW

File: features/config/TEMPLATE_complex.xml [hunk 1]
```diff
@@ -8,5 +8,6 @@
     <rules>
        <optlevelRule excludeOptlevelNameString="usmnone" />
        <optlevelRule GPUFeature="NOT double" excludeOptlevelNameString="gpu" />
+       <optlevelRule excludeOptlevelNameString="cuda" />
     </rules>
 </test>
```
Comment (yihwang-nv) [inline features/config/TEMPLATE_complex.xml]: New line

**Review by yihwang-nv — APPROVED** [review]

File: help_function/help_function.xml [hunk 2]
```diff
@@ -3,42 +3,42 @@
...
-    <test testName="assign_device_vector" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="move_device_vector" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="atomic_add_float" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="atomic_fetch_compare_inc" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="async_exception" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="dpct_device_info" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="max" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="memcpy3D" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="memcpy3D_2" configFile="config/TEMPLATE_help_function.xml" />
+    <test testName="array_copy" configFile="config/TEMPLATE_help_function_eanble_cuda_backend.xml" />
```
Comment (tangjj11) [inline help_function/help_function.xml]:
```suggestion
    <test testName="array_copy" configFile="config/TEMPLATE_help_function_enable_cuda_backend.xml" />
```

**Review by tangjj11 — CHANGES_REQUESTED** [review]

**Review by tangjj11 — APPROVED** [review]

File: help_function/help_function.xml [hunk 3]
```diff
@@ -3,42 +3,42 @@
...
-    <test testName="move_device_vector" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="atomic_add_float" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="atomic_fetch_compare_inc" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="async_exception" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="dpct_device_info" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="max" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="memcpy3D" configFile="config/TEMPLATE_help_function.xml" />
-    <test testName="memcpy3D_2" configFile="config/TEMPLATE_help_function.xml" />
+    <test testName="array_copy" configFile="config/TEMPLATE_help_function_enable_cuda_backend.xml" />
+    <test testName="assign_device_vector" configFile="config/TEMPLATE_help_function_enable_cuda_backend.xml" />
```
Comment (ShengchenJ) [inline help_function/help_function.xml]: Please keep the help function configuration xml. And skip the CUDA backend if needed.

File: run_test.py [hunk 4]
```diff
@@ -282,7 +282,7 @@ def test_single_case(current_test, single_case_config, workspace, module, suite_
     if single_case_config.option_rule_list and not is_option_supported(single_case_config.option_rule_list):
         append_msg_to_file(test_config.result_text, current_test + " Skip " + "\n")
         return True
-
+
```
Comment (ShengchenJ) [inline run_test.py]: Please remove the no need space.

**Review by ShengchenJ — CHANGES_REQUESTED** [review]

**Review by ShengchenJ — APPROVED** [review]

File: help_function/config/image_wrapper.xml [hunk 5]
```diff
@@ -7,6 +7,7 @@
     </files>
     <rules>
     <optlevelRule excludeOptlevelNameString="acc" />
+      <optlevelRule excludeOptlevelNameString="cuda" />
```
Comment (zhimingwang36) [inline help_function/config/image_wrapper.xml]: ### 
```suggestion
      <optlevelRule excludeOptlevelNameString="cuda-backend" />
```

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**